client/gutenboarding/store/types.ts

```diff
@@ -10,3 +12,26 @@ export enum SiteType {
 	STORE = 'store',
 	STORY = 'story',
 }
+
+export type Vertical = ApiVertical | UserVertical;
+
+export interface UserVertical {
+	is_user_input_vertical: true;
+	vertical_id: string;
+	vertical_slug: string;
+	vertical_name: string;
+	parent: string;
+	preview: string;
+	preview_styles_url: string;
+	synonyms: string[];
+}
+
+export interface ApiVertical {
+	is_user_input_vertical: false;
+	vertical_id: '';
+	vertical_slug: string;
+	vertical_name: string;
+	parent: string;
+	preview: '';
+	preview_styles_url: string;
+}
```

Comment:
There are predefined verticals and the user-provided vertical. This intends to represent both of them based on observed vertical data responses.